tds-prize-tool/src/rewards_earned.rs

```diff
@@ -0,0 +1,282 @@
+use crate::prize::{self, Winner, Winners};
+use solana_runtime::bank::Bank;
+use solana_sdk::account::Account;
+use solana_sdk::pubkey::Pubkey;
+use solana_stake_api::stake_state::StakeState;
+use solana_vote_api::vote_state::VoteState;
+use std::cmp::{max, min};
+use std::collections::HashMap;
+
+pub const STARTING_BALANCE_LAMPORTS: i64 = 0; // TODO
+
+const HIGH_BUCKET: &str = "Top 25% Bucket";
+const MID_BUCKET: &str = "Top 25-50% Bucket";
+const LOW_BUCKET: &str = "Top 50-90% Bucket";
+
+fn voter_stake_rewards(stake_accounts: HashMap<Pubkey, Account>) -> HashMap<Pubkey, u64> {
+    let mut voter_stake_sum: HashMap<Pubkey, u64> = HashMap::new();
+    for (_key, account) in stake_accounts {
+        if let Some(StakeState::Stake(_authorized, _lockup, stake)) = StakeState::from(&account) {
+            let stake_sum = voter_stake_sum
+                .get(&stake.voter_pubkey)
+                .cloned()
+                .unwrap_or_default();
+            voter_stake_sum.insert(stake.voter_pubkey, stake_sum + account.lamports);
```

Comment:
strictly speaking the staked amount is a function of epoch and stake_history, this simplification can only be made if we ignore warmup

Reply:
Yeah, this tool isn't calculating total stake. It's calculating the total of lamports attributed to a validator through the balances of stake accounts and vote accounts according to @mvines suggestion here https://github.com/solana-labs/solana/issues/6077#issuecomment-535564165